services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -329,12 +344,63 @@ open class ApplicationService(
             logger.info { "Application deleted, id=$id, alluid=$alluid userid=$userId" }
         }
 
-    private fun initAccessForApplication(
+    /** Creates access for all application contacts, excluding the current user. */
+    private fun initAccessOnInitialSend(
         application: ApplicationEntity,
         hanke: HankeEntity,
-        currentUserId: String
+        currentUserId: String,
+    ) {
+        if (featureFlags.isDisabled(Feature.USER_MANAGEMENT)) {
+            logger.info { "Feature ${Feature.USER_MANAGEMENT} disabled. No tokens created." }
+            return
+        }
+
+        val kayttaja = hankeKayttajaService.getKayttajaByUserId(hanke.id!!, currentUserId)
+        val contacts = application.applicationData.typedContacts(omit = kayttaja?.sahkoposti)
+
+        initAccess(
+            application = application,
+            hanke = hanke,
+            currentKayttaja = kayttaja,
+            currentUserId = currentUserId,
+            contacts = contacts
+        )
+    }
+
+    /** Creates access for new application contacts. Email address used for comparison. */
+    private fun initAccessOnAlluUpdate(
```

Comment:
I think access can be initialized just once. Calling this init feels odd.

Reply:
Renamed init to provide.